types/service/database_service_configuration.go

```diff
@@ -218,13 +343,30 @@ type DatabaseTlsAuthConfiguration struct {
 	TlsConfig
 }
 
+func (config DatabaseTlsAuthConfiguration) Validate() error {
+	if config.Username == "" || config.Password == "" {
```

Comment:
prefer `if` per condition -- to be able to provide more accurate error messages. e.g. `username and password are required` doesnt tell the user whether the username OR password was missing.